framework/wazuh/core/cluster/hap_helper/hap_helper.py

```diff
@@ -432,14 +434,26 @@ async def start(cls):
             )
 
             wazuh_dapi = WazuhDAPI(
-                tag=tag, excluded_nodes=configuration['wazuh']['excluded_nodes'],
+                tag=tag,
+                excluded_nodes=configuration['wazuh']['excluded_nodes'],
             )
 
             helper = cls(proxy=proxy, wazuh_dapi=wazuh_dapi, tag=tag, options=configuration['hap_helper'])
 
             await helper.initialize_proxy()
+
+            if helper.proxy.hard_stop_after is not None:
+                helper.logger.info(
+                    'Ensuring only exists one HAProxy process. '
+                    f'Sleeping {helper.proxy.hard_stop_after}s before start...'
+                )
+                await sleep(helper.proxy.hard_stop_after)
```

Comment:
The `hard_stop_after` value could be smaller than the reconnection retry, so `await helper.manage_wazuh_cluster_nodes()` would be run before all workers reconnect. Maybe the sleep value should be the maximum between `'connection_retry' + 2` or the actual hard_stop_after.